program/src/instruction.rs

```diff
@@ -114,11 +124,27 @@ pub(crate) fn decode_instruction_data<T: Pod>(input_with_type: &[u8]) -> Result<
 
 /// Create a `SlashingInstruction::DuplicateBlockProof` instruction
 pub fn duplicate_block_proof(
+    reporter: &Pubkey,
     proof_account: &Pubkey,
     instruction_data: &DuplicateBlockProofInstructionData,
 ) -> Instruction {
-    let mut accounts = vec![AccountMeta::new_readonly(*proof_account, false)];
-    accounts.push(AccountMeta::new_readonly(sysvar::instructions::id(), false));
+    let (pda, _) = Pubkey::find_program_address(
```

Comment:
Is this the recommend way to do this? Both the caller and the program know the seeds for the PDA and derive it independently.

Reply:
Got it, ended up pulling out a helper to lib.rs